### DISCUSSION TITLE
High Availability?

### DISCUSSION THREAD

**Discussion by danieledwardgeorgehitchcock** [Koenkk/zigbee2mqtt#7142] [General]
Just wondering if it would be theoretically possible to run Zigbee2MQTT in a high availability environment?

I have a server cluster of 4 nodes, each are set to high availability failover.

Would it be possible to run identical Zigbee2MQTT coordinator hardware on each of the nodes, have the Zigbee2MQTT databases and config in a centralised storage location, and set it up in such a way, that the node Zigbee2MQTT is running on will coordinate the zigbee network?

**Comment by NineEighty** [comment]
Did you get anywhere with this? Im also wondering the same thing.

**Comment by chemadh** [comment]
I've been playing around this concept during some months... please find my prototype approach in the following repository:

https://github.com/chemadh/zigbee2mqtt_ha

Any feedback is welcome!

**Reply by danieledwardgeorgehitchcock** [reply]
This looks like a really solid Proof of Concept and is one I am keen to start investigating / tinkering with!

It would be good to get a tighter integration with Zigbee2MQTT and maybe containerize the components so that some cool features can be done in an orchestration service such as Kubernetes with its HA failover handling. 

It would be great to have the flexibility to have n* failover nodes instead of just the two for instance, I have a 4 machine High Availability setup - I'd like to run the same hardware and services on each of the nodes but just have Z2M active on a machine which gains quorum over the others.
